lib/media/media_source_engine.js

```diff
@@ -1333,6 +1366,173 @@ shaka.media.MediaSourceEngine = class {
     return segment;
   }
 
+  /**
+   * Prepare the SourceBuffer to parse a potentially new type or codec.
+   *
+   * @param {shaka.util.ManifestParserUtils.ContentType} contentType
+   * @param {string} newFullMimeType
+   * @private
+   */
+  change_(contentType, newFullMimeType) {
+    const ContentType = shaka.util.ManifestParserUtils.ContentType;
+    if (contentType === ContentType.TEXT) {
+      shaka.log.debug(`Change not supported for ${contentType}`);
+      return;
+    }
+    shaka.log.debug(`Change Type: ${contentType}`);
+    shaka.log.debug(
+        `${this.sourceBufferTypes_[contentType]} -> ${newFullMimeType}`);
+    if (shaka.media.Capabilities.isChangeTypeSupported()) {
+      this.sourceBuffers_[contentType].changeType(newFullMimeType);
```

Comment:
It is not guaranteed that any call to changeType will work merely because the old and new types are both supported separately.  In particular, hardware decoders and EME can have an impact on this capability.  You need to use MediaCapabilities to determine which changeType() calls will work, and ideally incorporate that into the filtering logic.